src/Main/Views/Income/ChartD3js/ChartD3.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect } from 'react';
+import * as d3 from "d3";
+
+import chartData from "../data/chartData";
+
+import "./ChartD3.css";
+
+
+
+const ChartD3 = () => {
+
+    let data = Object.values(Object.entries(chartData)[0][1]);
+    data = data.map(el => { return Math.round(el) });
+    let labels = []
+    labels.length = 26;
+    labels.fill(null);
+    labels = labels.map((el, idx) => { return el = idx });
+
+    useEffect(() => {
+        const width = 850;
+        const height = 450;
+        const padding = 40;
+
+        const xScale = d3.scaleBand()
+            .domain(labels)
+            .range([padding, width - padding])
+            .padding([0.1]);
+        const xAxis = d3.axisBottom(xScale);
+
+        const yScale = d3.scaleLinear()
+            .domain([d3.min(data), d3.max(data)])
+            .range([height - padding, 0]);
+        const yAxis = d3.axisLeft(yScale);
+
+        //the principal svg
+        const svg = d3.select("#ChartD3")
+            .append("svg")
+            .attr("height", height)
+            .attr("width", width);
+
+        //bars
+        svg.selectAll("rect")
+            .data(data)
+            .enter()
+            .append("rect")
+            .attr("x", (d, i) => xScale(labels[i]))
+            .attr("y", (d) => yScale(d))
+            .attr("height", (d, i) => (height - padding) - yScale(d))
+            .attr("width", xScale.bandwidth())
+            .attr("class", "bar")
+            .append("title")
+            .text(d => `PlueValue: ${d} euros`);
+
+        // label for bottom axis
+        svg.append("text")
+            .attr("x", "50%")
+            .attr("y", (d) => height - 2 + "px")
```

Comment:
Tutaj tylko taka mała "ciekawostka". Aktualnie tekst zaczyna się w połowie, 50% tej osi. Przy dłuższych tekstach byłoby widać wyraźnie, że nie jest do końca wypośrodkowany. By to zrobić możesz użyć:
``` .attr("text-anchor", "middle")```
Wtedy to środek tekstu pozycjonowany jest na te 50% :)

Reply:
@nettecode Hmm, właśnie spróbowałam "anchorować", ale chyba początek mojego tekstu wyznacza się gdzieś indziej, niż początek mojego svg, przez co środek nie wychodzi tam gdzie ma 🤔 Chociaż "start" wyznacza się na początku svg. Dlatego na razie zostawiam 50%, póki nie dojdę, dlaczego middle nie jest w middle.